examples/ProjPlaneCurve.jl

```diff
@@ -0,0 +1,534 @@
+  if dim(PP) != length(p)
+     error("Not the right size")
+  end
+  m = push!(p, 1)
+  q = Oscar.Geometry.ProjSpcElem(PP, m)
+  return q
+end
+
+################################################################################
+# Compute the intersection of two curves. The output is a list: the first
+# element the affine plane curve defined by the common components of the two
+# curves (or is empty if no common component), the second element is the list
+# of intersection points. Some might be contained in the common component too.
+
+@doc Markdown.doc"""
+curve_intersect(PP::Oscar.Geometry.ProjSpc{S}, C::ProjPlaneCurve{S}, D::ProjPlaneCurve{S}) where S <: FieldElem
+> Returns a list whose first element is the projective plane curve defined by the gcd of F and G, the second element is the list of the remaining intersection points when the common components are removed from C and D (the points are in PP).
+"""
+function curve_intersect(PP::Oscar.Geometry.ProjSpc{S}, C::ProjPlaneCurve{S}, D::ProjPlaneCurve{S}) where S <: FieldElem
+  G = gcd(C.eq.f, D.eq.f)
+  R = parent(C.eq)
+  CC = []
+  if isone(G) == false
+     # We divide by the gcd to get curves without common components.
+     F = R(div(C.eq.f, G))
+     H = R(div(D.eq.f, G))
+     push!(CC, ProjPlaneCurve(G))
+  else
+     F = C.eq
+     H = D.eq
+  end
+  Pts = []
+  r, (X, Y) = PolynomialRing(R.R.base_ring, ["X", "Y"])
+  Fa = dehomogenization(r, F, 3)
+  Ha = dehomogenization(r, H, 3)
+  if isconstant(Fa) == false && isconstant(Ha) == false
+     Ca = AffinePlaneCurveModule.AffinePlaneCurve(Fa)
+     Da = AffinePlaneCurveModule.AffinePlaneCurve(Ha)
+     L = AffinePlaneCurveModule.curve_intersect(Ca, Da)
+     for p in L[2]
+        push!(Pts, Array_to_ProjSpcElem(PP, p.coord))
+     end
+  end
+  rr, (x) = PolynomialRing(R.R.base_ring, ["x"])
+  phi = hom(R.R, rr, [gens(rr)[1], rr(1), rr(0)])
+  phiF = phi(F)
+  phiH = phi(H)
+  g = gcd(phiF, phiH)
+  f = factor(g)
+  ro = []
+  for h in keys(f.fac)
+     if total_degree(h) == 1
+        f = h//lc(h)
+        push!(ro, -f + gens(rr)[1])
+     end
+  end
+  for y in ro
+     push!(Pts, Oscar.Geometry.ProjSpcElem(PP, [lc(y), R.R.base_ring(1), R.R.base_ring(0)]))
+  end
+  if evaluate(F, [1, 0, 0]) == 0 && evaluate(H, [1, 0, 0]) == 0
+     push!(Pts, Oscar.Geometry.ProjSpcElem(PP, [R.R.base_ring(1), R.R.base_ring(0), R.R.base_ring(0)]))
+  end
+  return [CC,Pts]
+end
+
+################################################################################
+# without specifying the projective space.
+
+@doc Markdown.doc"""
+curve_intersect(C::ProjPlaneCurve{S}, D::ProjPlaneCurve{S}) where S <: FieldElem
+> Returns a list whose first element is the projective plane curve defined by the gcd of F and G, the second element is the list of the remaining intersection points when the common components are removed from C and D (the points are in a new projective space).
```

Comment:
```suggestion
    curve_intersect(C::ProjPlaneCurve{S}, D::ProjPlaneCurve{S}) where S <: FieldElem

Return a list whose first element is the projective plane curve defined by the gcd of `F` and `G`, and whose second
element is the list of the remaining intersection points when the common components are removed from `C` and `D`
(the points are in a new projective space).
```
Alternatively, you might consider merging this docstring with that of the previous method, with a signature like 
```
curve_intersect([PP::Oscar.Geometry.ProjSpc{S}], C::ProjPlaneCurve{S}, D::ProjPlaneCurve{S}) where S <: FieldElem
```
Where the `[...]` brackets means optional. And then:
"... (the points are in `PP` if specified, or in a new projective space otherwise)."